app/controllers/statement_loaders_controller.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+class StatementLoadersController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  def new
+    @organization = Organization.find(params[:id])
+  end
+
+  def show; end
+
+  def extract_tax
+    raise 'Exception Upload CSV' if params[:my_file].blank?
+    tmp = params[:my_file].tempfile
+    csv_text = File.read(tmp)
+    csv = CSV.parse(csv_text, headers: true)
+    array_obj = CsvParser.new(csv).fetch
+    @data_statement = select_type_tax(params[:id]).new(array_obj)
+    render :show, data: @data_statement
+  end
+
+  def redirect_taxes
+    redirect_to taxes_path
+  end
+
+  private
+
+  def select_type_tax(organization_id)
+    @organization = Organization.find(organization_id)
+    case @organization.taxation_form.name
+    when 'УСН'
+      CalculateExtractForUsn
+    when 'Единый налог'
+      CalculateExtractForSingle
+    end
+  end
+end
+
+class CsvParser
```

Comment:
move these classes to separate entities, add factories to fetch necessary objects

you should have BgpbCsvParser and calculator for the case when you work with dollars.